unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,26 @@
+#lang racket/base
+
+(provide hash-slice-lens)
+
+(require lens/base/main)
+(module+ test
+  (require rackunit))
+
+(define (hash-slice-lens . ks)
+  (define (get hsh)
+    (for/hash ([k (in-list ks)])
+      (values k (hash-ref hsh k))))
+  (define (set hsh hsh2)
+    (define hsh1
+      (for/fold ([hsh hsh]) ([k (in-list ks)])
+        (hash-remove hsh k)))
+    (for/fold ([hsh hsh1]) ([(k v) (in-hash hsh2)])
+      (hash-set hsh k v)))
+  (make-lens get set))
+
+(module+ test
+  (check-equal? (lens-view (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'd 4))
```

Comment:
What do you mean? It's still pure, has get-set consistency, and has set-get consistency, right?
(a few seconds later)
Oh, wait, if the context function gets a hash that has keys that were in the original hash, then it might not? I'm not sure.